paper-api/src/main/java/io/papermc/paper/block/Lidded.java

```diff
@@ -0,0 +1,37 @@
+package io.papermc.paper.block;
+
+import org.bukkit.block.TileState;
+import org.jspecify.annotations.NullMarked;
+
+@NullMarked
+public interface Lidded extends TileState {
+
+    /**
+     * Gets the current state of the block, respecting the lidded mode.
+     *
+     * @return the effective lid state
+     */
+    LidState getEffectiveLidState();
+
+    /**
+     * Gets how the lid would be without any lidded mode, based on players interacting with the block.
```

Comment:
TODO: replace with smth like

```suggestion
     * Gets how the lid would be with {@link LidMode.DEFAULT}
     * I.E. based on players interacting with the block.
```